src/test/scala/testCBO/CBOAnalysisTest.scala

```diff
@@ -0,0 +1,59 @@
+package org.tud.sse.metrics
+package testCBO
+
+import impl.group3.CBOAnalysis
+import testutils.AnalysisTestUtils
+
+import org.scalatest.{FlatSpec, Matchers}
+
+import java.io.File
+
+class CBOAnalysisTest extends FlatSpec with Matchers{
+
+  "The CBOAnalysis" must "calculate valid results for single JARs" in {
+    /**
+     * tested with default jar file, and our own file(converted from source code). result is a list of tuple which refers to diffent classfile
+     */
+    val fileToTest = new File(getClass.getResource("/demo/utils-3.5-beta76.jar").getPath)
+
+    val analysisToTest = new CBOAnalysis()
+
+    val appConfig = new ApplicationConfiguration(inputFilePath = "", treatFilesAsLibrary = true, outFileOption = None,
+      opalLoggingEnabled = false, batchModeEnabled = Some(false), excludedAnalysesNames = List(), includedAnalysesNames = List(),
+      excludeJreClasses = true, additionalClassesDir = None, loadAdditionalClassesAsInterface = false)
+
+    val result = AnalysisTestUtils.runSingleFileAnalysis(analysisToTest, fileToTest, appConfig, Map.empty[Symbol, Any])
+
+    assert(result.size == 1)
+
+    val metricResult = result.head
+
+    assert(metricResult.success)
+    assert(metricResult.metricValues.nonEmpty)
+    assert(metricResult.metricValues.exists(value => value.metricValue == 0.0))
```

Comment:
Die Existenz einer Wert in `metricResult` reicht nicht um zu prüfen dass, der richtige Wert für das richtige Entity gegeben ist